apache-pulsar/src/main/java/io/debezium/examples/apache/pulsar/PulsarProducer.java

```diff
@@ -0,0 +1,171 @@
+        valueConverter.configure(config.asMap(), false);
+    }
+
+    private Producer<String> getProducer(String topic) {
+        String topicFormat = propConfig.getProperty("pulsar.topic");
+        String topicURI = MessageFormat.format(topicFormat, topic);
+        Producer<String> producer = producerHashMap.get(topic);
+
+        if (producer == null) {
+            try {
+                producer = client.newProducer(Schema.STRING)
+                        .topic(topicURI)
+                        .create();
+            }
+            catch (PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+
+            producerHashMap.put(topic, producer);
+        }
+
+        return producer;
+    }
+
+    @Override
+    public void run() {
+        final EmbeddedEngine engine = EmbeddedEngine.create()
+                .using(config)
+                .using(this.getClass().getClassLoader())
+                .using(Clock.SYSTEM)
+                .notifying(this::sendRecord)
+                .build();
+
+        try {
+            client = PulsarClient.builder()
+                    .serviceUrl(propConfig.getProperty("pulsar.broker.address"))
+                    .build();
+        }
+        catch (PulsarClientException e) {
+            throw new RuntimeException(e);
+        }
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(engine);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            logger.info("Requesting embedded engine to shut down");
+            try {
+                engine.stop();
+
+                producerHashMap.forEach((topic, producer) -> {
+                    logger.info(String.format("Closing producer for topic %s", topic));
+                    try {
+                        producerHashMap.get(topic).close();
+                    }
+                    catch (PulsarClientException e) {
+                        logger.error("Couldn't close producer", e);
+                    }
+                    logger.info("Producer Closed");
+
+                });
+                client.close();
+            }
+            catch (PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+        }));
+
+        awaitTermination(executor);
+    }
+
+    /**
+     * For Every record this method will be invoked* @param record
+     */
+    private void sendRecord(SourceRecord record) {
+        final byte[] payload = valueConverter.fromConnectData("dummy", record.valueSchema(), record.value());
+        final byte[] key = keyConverter.fromConnectData("dummy", record.keySchema(), record.key());
+        logger.debug("Publishing Topic --> " + record.topic());
+        logger.debug("Key -->" + new String(key));
+        logger.debug("Payload --> " + new String(payload));
+        Producer<String> producer;
+
+        producer = getProducer(record.topic());
+
+        while (true) {
```

Comment:
I do not think the `while` loop is necessary here

Reply:
I think that's fine.